bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/config/KNXGenericBindingProvider.java

```diff
@@ -380,10 +386,18 @@ protected KNXBindingConfig parseBindingConfigString(Item item, String bindingCon
                     throw new BindingConfigParseException("Only one readable GA allowed.");
                 }
 
-                Class<? extends Type> typeClass = item.getAcceptedCommandTypes().size() > 0
-                        ? item.getAcceptedCommandTypes().get(i)
-                        : item.getAcceptedDataTypes().size() > 1 ? item.getAcceptedDataTypes().get(i)
-                                : item.getAcceptedDataTypes().get(0);
+                Class<? extends Type> typeClass = null;
+                if( item.getAcceptedCommandTypes().size() > i ) {
+                  typeClass = item.getAcceptedCommandTypes().get(i);
+                }
+                else {
+                  if( item.getAcceptedDataTypes().size() > i ) {
+                    typeClass = item.getAcceptedDataTypes().get(i);
+                  }
+                  else {
+                    typeClass = item.getAcceptedDataTypes().get(0);
+                  }
+                }
```

Comment:
I wonder why this code is indexing into the list of the item's accepted data and command types?  I know this existed before this PR, but doesn't it seem wrong that the KNX binding would expect to have knowledge of which States and Commands exist at certain indices in the lists?  This would be a great opportunity to figure out how to disconnect this binding from any expectations of which State and Command classes exist at specific indices in the list.